rasa/server.py

```diff
@@ -615,6 +623,61 @@ def _get_events_from_request_body(request: Request) -> List[Event]:
 
         return response.json(response_body)
 
+    @app.post("/conversations/<conversation_id>/trigger_intent")
+    @requires_auth(app, auth_token)
+    @ensure_loaded_agent(app)
+    async def trigger_intent(request: Request, conversation_id: Text) -> HTTPResponse:
+        request_params = request.json
+
+        intent_to_trigger = request_params.get("name")
+        entities = request_params.get("entities", [])
+
+        if not intent_to_trigger:
+            raise ErrorResponse(
+                400,
+                "BadRequest",
+                "Name of the intent not provided in request body.",
+                {"parameter": "name", "in": "body"},
+            )
+
+        verbosity = event_verbosity_parameter(request, EventVerbosity.AFTER_RESTART)
+
+        try:
+            async with app.agent.lock_store.lock(conversation_id):
+                tracker = await get_tracker(
+                    app.agent.create_processor(), conversation_id
+                )
+                output_channel = _get_output_channel(request, tracker)
+                if intent_to_trigger not in app.agent.domain.intents:
+                    raise ErrorResponse(
+                        404,
+                        "NotFound",
+                        f"The intent {trigger_intent} does not exist in the domain.",
+                    )
+                await app.agent.trigger_intent(
+                    intent_name=intent_to_trigger,
+                    entities=entities,
+                    output_channel=output_channel,
+                    tracker=tracker,
+                )
+        except ErrorResponse:
+            raise
+        except Exception as e:
+            logger.debug(traceback.format_exc())
+            raise ErrorResponse(
+                500, "ConversationError", f"An unexpected error occurred. Error: {e}"
+            )
+
+        tracker = await get_tracker(app.agent.create_processor(), conversation_id)
```

Comment:
you can just use the old tracker, right? you don't have to get it again. All changes are done within the object, so `tracker` should still refer to the latest imo. Best double check 🤔